apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -90,12 +90,13 @@ type RootDeviceHints struct {
 }
 
 // BootMode is the boot mode of the system
-// +kubebuilder:validation:Enum=UEFI;legacy
+// +kubebuilder:validation:Enum=UEFI;UEFI secure boot;legacy
 type BootMode string
 
 // Allowed boot mode from metal3
 const (
 	UEFI            BootMode = "UEFI"
+	SecureUEFI      BootMode = "UEFI secure boot"
```

Comment:
nit, but I'm not sure if spaces are typically used in enumerated values like this - I wonder if it should be something like `SecureBootUEFI` or just `SecureBoot` if we know that only works with UEFI?

@dhellmann any thoughts on that?

Reply:
Let's avoid spaces in the enum. I don't think they're forbidden, but as a style thing I haven't seen them enough to say they're "liked" generally.

In terms of the actual name to use, I don't have an opinion.

Reply:
`UEFISecureBoot` resembles the canonical name of the feature the most, I'll use it.

Reply:
The only problem (that led me to this) is that `legacy` is low-cased.

Reply:
What do kubernetes (or OpenShift?) API guidelines say about enum values? I think they want capital letters, so we got `legacy` wrong.